### PR TITLE
Final adjustments before publishing stampit v3.

### PR DESCRIPTION
**Pull request by koresar**
After this PR I'm publishing stampit v3. Please, take an extra care.
* Spelling
* Ambiguous text
* Something is missing
* Something is redundant
* Basically, tell me anything you would improve for better readability and clearness

Thanks everyone!

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -273,237 +251,315 @@ Object.assign(stamp, {
...
+* All chaining methods return new stamps instead of self-mutating this stamp.
+* `state()` always shallow merge properties. It was not doing so in a single rare case.
+* Instead of factory arguments the `enclose()` functions now receive the following object `{ instance, stamp, args }`.
+
+##### New features
+* `stampit()` now receives options object (`{methods,refs,init,props,static}`) instead of multiple arguments.
+* Instead of factory arguments the `enclose()` functions now receive the following object `{ instance, stamp, args }`.
+* New `stamp.props()` method for deeply merged state.
+* New `stamp.statics()` method which add properties to **stamp**, not an object.
+* `state` deprecated. `refs` must be used instead.
```
Comment (abhisekp) [inline docs/API.md]: `refs` are deprecated too

Comment (koresar) [inline docs/API.md]: That is about stampit v2. The `refs` were not deprecated at that time.

File: docs/API.md [hunk 2]
```diff
@@ -115,91 +144,76 @@ const stamp = stampit().methods({
 stamp().amplify('BADF00D'); // value BADF00D is incorrect
 ```
 
-### stamp.refs()
+### stamp.props() and stamp.props() and stamp.properties()
```
Comment (abhisekp) [inline docs/API.md]: *huh*?

2 stamp.props()
and only 1 stamp.properties? :expressionless:

File: docs/API.md [hunk 3]
```diff
@@ -273,237 +251,317 @@ Object.assign(stamp, {
...
+```
+Became:
+```js
+const Stamp = stampit({init(arg, {instance, stamp, args}) {
+  console.log(this);
+}});
+Stamp({foo: 'bar'}); // {}
+```
+
+* A stamp's metadata is now stored in the `stamp.compose` object. Previously in the `stamp.fixed`.
```
Comment (abhisekp) [inline docs/API.md]:
```diff
- Previously in the `stamp.fixed`.
+ Previously it was stored in `stamp.fixed` object.
```

Comment (koresar) [inline docs/API.md]: Done!

File: docs/API.md [hunk 4]
```diff
@@ -1,3 +1,39 @@
...
+    - [Pass multiple objects into all methods and functions](#pass-multiple-objects-into-all-methods-and-functions)
+- [Breaking changes](#breaking-changes)
+  - [Stampit v2](#stampit-v2)
+  - [Stampit v3](#stampit-v3)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+
 # Stampit API
```
Comment (abhisekp) [inline docs/API.md]: Move the heading to the top before ToC

File: docs/API.md [hunk 5]
```diff
@@ -115,91 +180,76 @@ const stamp = stampit().methods({
...
 
 Functions passed into `.init()` are called any time an
 object is instantiated. That happens when the stamp function
 is invoked, or when the `.create()` method is called.
 
-Each function receives the following object as the first argument:
+If an initializer returns a non-undefined value then it becomes the factory 
+instantiated value.
+
+Each function receives the following object as the second argument:
```
Comment (FredyC) [inline docs/API.md]: No mention of what is the first argument. It's probably understandable from example below, but might be good to say: "Each function receives first argument of called factory function and second object argument like this:"

File: docs/API.md [hunk 6]
```diff
@@ -115,91 +180,76 @@ const stamp = stampit().methods({
...
 ```
 {
-  instance,
-  stamp,
-  args
+  instance, // The same as `this`
```
Comment (FredyC) [inline docs/API.md]: Perhaps mention what is this good for - ES6 fat arrows. I remember I was confused about it a lot.

File: docs/API.md [hunk 7]
```diff
@@ -273,237 +287,315 @@ Object.assign(stamp, {
...
-### stampit.mixin(), .extend(), .mixIn(), .assign()
+Take n objects and add all its properties to the stamp's metadata.
 
-Aliases to `Object.assign()`. Deprecated.
+* `@return {Object} stamp` The new stamp based on the original `this` stamp.
 
+#### Examples
 
-### stampit.isStamp(obj)
+Use metadata in intializers:
```
Comment (FredyC) [inline docs/API.md]: initializers

File: docs/API.md [hunk 8]
```diff
@@ -273,237 +287,315 @@ Object.assign(stamp, {
...
+```js
+const stamp = stampit()
+.conf({addFactorSetter: false})
+.init((opts, {stamp}) => {
+  let factor = opts.factor || 1;
+  instance.getFactor = () => factor;
```
Comment (FredyC) [inline docs/API.md]: You are missing `instance` in arguments

File: docs/API.md [hunk 9]
```diff
@@ -273,237 +287,315 @@ Object.assign(stamp, {
...
-Note that if you use this feature, it is **not safe** to compose
-the resulting stamp with other stamps willy-nilly, because if two
-different stamps depend on the argument passing feature, the arguments
-will probably clash with each other, producing very unexpected results.
+### stamp.deepConf() and stamp.deepConfiguration()
 
- * @param  {Function} Constructor
- * @return {Function} A composable stampit factory (aka stamp).
+Same as `stamp.conf()` and `stamp.configuration()` but deeply merges the
+provided objects.
```
Comment (FredyC) [inline docs/API.md]: Might be worth mentioning here that result is accessible in `stamp.compose.deepConfiguration`. That's another difference and might cause confusion people wondering why they don't see `configuration` correctly.

File: docs/API.md [hunk 10]
```diff
@@ -273,237 +287,315 @@ Object.assign(stamp, {
...
 
-## Examples
 
-### Pass multiple objects into .methods(), .refs(), .init(), props(), .static(), or .compose().
+### stamp.compose([arg1] [,arg2] [,arg3...])
 
-Every fluent method of stampit can receive multiple arguments.
-The properties from later arguments in the list will override the same named properties of previously passed in objects. 
+Take one or more stamp or stamp descriptors and
+combine them with `this` stamp to produce and return a new factory object.
```
Comment (FredyC) [inline docs/API.md]: "and return a new stamp" ?

File: docs/API.md [hunk 11]
```diff
@@ -273,237 +287,315 @@ Object.assign(stamp, {
...
+Just like calling `stamp()`, `stamp.create()` invokes the stamp
+and returns a new object instance. 
 
 ```js
-  const obj = stampit().init(function () {
-    console.log(this);
-  }, function () {
-    console.log(this); // same as above
-  }).create();
+const stamp = stampit().init((opts, {args}) => { console.log(args); });
```
Comment (FredyC) [inline docs/API.md]: `console.log(...args)` to get the desired output

File: docs/API.md (hunk 9 above)
Comment (abhisekp) [inline docs/API.md]: ...and also that the statement is two lines.

**Pull request merged**